### ISSUE TITLE
Progress handler not being called for succesful SFTP.get() of 0 byte files

### ISSUE THREAD

**Issue by geraldlnj** [ronf/asyncssh#314]
In regular SFTP.get() of files, the progress handler seems to be called in _SFTPFileCopier.run_task(). However, if the file size is 0 bytes, even though the file is successfully downloaded, the progress handler is never called.

This seems to be because of _SFTPParallelIO._start_tasks(), which calls _SFTPFileCopier.run_task() in line 399:

```
while self._bytes_left and len(self._pending) < self._max_requests:
            size = min(self._bytes_left, self._block_size)
            task = asyncio.ensure_future(self.run_task(self._offset, size))
```
If self._bytes_left is 0 due to zero byte files, self.run_task is never called, so the progress handler isn't called as well.

**Comment by ronf** [comment]
Good point. In all other cases, it will do at least one read(), and even if that returns less than a full block's worth of data it'll still call the progress handler after copying those bytes. However, if the source file is zero length, it will never do any I/O after opening the files, and so the progress handler is never run as you said. It seems reasonable to make a special call to it in that case, I guess, reporting that 0 bytes were copied. I'll look into that.

**Comment by geraldlnj** [comment]
What if we create `self.handler = None` in the _SFTPParallelIO parent class, then set `self.handler = handler` or `self.handler = progress_handler` in it's subclasses? Then we can call the handler from _SFTPParallelIO.

**Comment by ronf** [comment]
I'm thinking this could possibly be handled in the start() method of the subclasses.

**Comment by ronf** [comment]
The ParallelIO class is used for breaking a single large read/write call into multiple smaller calls, without any progress handler being involved in that case. So, I don't think it should go in that parent class.

**Comment by geraldlnj** [comment]
> I'm thinking this could possibly be handled in the start() method of the subclasses.

True, originally I was thinking that this wouldn't be good if there were more subclasses incoming, but I guess there won't be any more major additions to the subclasses

**Comment by ronf** [comment]
Ok - commit 59c352c is now available in the "develop" branch with what should be a fix for this. I also found and fixed a similar issue in the SCP case, and updated the unit tests to verify the new behavior. Thanks for catching this!

**Comment by geraldlnj** [comment]
No problem, thanks for helping to fix!

**Comment by ronf** [comment]
This fix is now available in AsyncSSH 2.4.2.
